libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -716,7 +716,13 @@ - (void)resumeSyncUpOneEntry:(SFSyncState*)sync
             [target updateOnServer:objectType objectId:objectId fields:fields completionBlock:completeBlockUpdate failBlock:failBlockUpdate];
             break;
         case SFSyncUpTargetActionDelete:
-            [target deleteOnServer:objectType objectId:objectId completionBlock:completeBlockDelete failBlock:failBlockDelete];
```

Comment:
Don't go to server when deleting locally created records (already on Android: https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/1159/commits/e2f720708069ecc083718f2e716f7083ffbed745)